python/seldon_core/batch_processor.py

```diff
@@ -316,23 +320,102 @@ def _send_batch_predict_multi_request(
     """
 
     indexes = [x[0] for x in input_data]
-    instance_ids = [x[1] for x in input_data]
+
+    seldon_puid = input_data[0][1]
+    instance_ids = [f"{seldon_puid}-item-{n}" for n, _ in enumerate(input_data)]
+    loaded_data = [json.loads(data[2]) for data in input_data]
 
     predict_kwargs = {}
     tags = {
         "batch_id": batch_id,
     }
-
     predict_kwargs["meta"] = tags
-    predict_kwargs["headers"] = {"Seldon-Puid": instance_ids[0]}
+    predict_kwargs["headers"] = {"Seldon-Puid": seldon_puid}
 
     try:
-        # Initialise concatenated array for data
-        loaded = [json.loads(raw_data[2]) for raw_data in input_data]
-        concat = np.concatenate(loaded)
-        predict_kwargs["data"] = concat
+        # Process raw input format
+        if data_type == "raw":
+            input_tags = [d.get("meta", {}).get("tags", {}) for d in loaded_data]
```

Comment:
May be worth just making a note that we ignore anything inside meta outside tags, I think this makes sense, but just in case someone expects (for some strange reason) other fields like metrics not being carried out

Reply:
Docstring comment here?

Reply:
primarily docs in batch page in seldon docs

Reply:
This should be documented now